### PR TITLE
fix: exception handling at event waiter

### PR DESCRIPTION
**Pull request by maw136**
Waiter - added manual cancellation token source used to cancel Waits by failing us

- useful for RunAndWait... scenarios
- reused in navigation scenarios as well

Issue: https://github.com/microsoft/playwright-dotnet/issues/2530

### CODE REVIEW

**Comment by maw136** [comment]
Those tests that fail are a little random at it. Locally all windows-based tests have passed... weird.

**Review by mxschmitt — COMMENTED** [review]
Thanks for your contribution. Please extract all the unrelated changes, like renamings, re-enabled tests, added tests which are already passing etc. in separate PRs, otherwise we are not able to review such a huge change.

**Comment by maw136** [comment]
OK, I am separating the cleanup changes to other PRs

**Comment by maw136** [comment]
Extracted:
https://github.com/microsoft/playwright-dotnet/pull/2542
https://github.com/microsoft/playwright-dotnet/pull/2540
https://github.com/microsoft/playwright-dotnet/pull/2538

**Comment by maw136** [comment]
I will rebase then to clean this PR

**Comment by maw136** [comment]
@mxschmitt any additional suggestions on splitting? IMHO the rest is pretty much tightly coupled.

**Comment by maw136** [comment]
How can I assist to merge and release the changes?

File: src/Playwright/Core/Page.cs [hunk 1]
```diff
@@ -30,6 +30,7 @@
 using System.Linq;
 using System.Text.Json;
 using System.Text.RegularExpressions;
+using System.Threading;
```
Comment (mxschmitt) [inline src/Playwright/Core/Page.cs:33]: revert this line

File: src/Playwright/Helpers/TaskHelper.cs [hunk 2]
```diff
@@ -204,6 +204,6 @@ private static async Task<bool> TimeoutTask(Task task, TimeSpan timeout)
 
     public static void IgnoreException(this Task task)
     {
-        _ = task.ContinueWith(t => t.Exception.Handle(_ => true), CancellationToken.None, TaskContinuationOptions.OnlyOnFaulted, TaskScheduler.Default);
+        _ = task.ContinueWith(t => t.Exception!.Handle(_ => true), CancellationToken.None, TaskContinuationOptions.OnlyOnFaulted, TaskScheduler.Default);
```
Comment (mxschmitt) [inline src/Playwright/Helpers/TaskHelper.cs:207]: revert this line

File: src/Playwright/Core/Frame.cs [hunk 3]
```diff
@@ -198,11 +198,43 @@ public async Task WaitForLoadStateAsync(LoadState? state = default, FrameWaitFor
...
+    private async Task<IResponse> WaitForNavigationInternalAsync(
+        Waiter waiter,
+        string url,
+        Func<string, bool> urlFunc,
+        Regex urlRegex,
+        string urlString,
+        WaitUntilState? waitUntil)
+    {
+        WaitUntilState waitUntil2 = waitUntil ?? WaitUntilState.Load;
+        string urlString2 = !string.IsNullOrEmpty(url) ? url! : urlString!;
```
Comment (mxschmitt) [inline src/Playwright/Core/Frame.cs:237]: maybe call it waitUntilNormalized and urlStringNormalized.

**Review by mxschmitt — APPROVED** [review]
lgtm, just nits.

File: src/Playwright/Core/Page.cs (hunk 1 above)
Comment (maw136) [reply src/Playwright/Core/Page.cs:33]: Good catch

**Pull request merged**